rest/proxy/proxy.go

```diff
@@ -0,0 +1,118 @@
+package proxy
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+
+	"github.com/fabric8-services/fabric8-wit/jsonapi"
+	"github.com/fabric8-services/fabric8-wit/log"
+
+	"github.com/goadesign/goa"
+	"github.com/pkg/errors"
+)
+
+// RouteHTTP uses a reverse proxy to route the http request to the scheme, host provided in targetHost
+// and path provided in targetPath.
+func RouteHTTPToPath(ctx context.Context, targetHost string, targetPath string) error {
+	return route(ctx, targetHost, &targetPath)
+}
+
+// RouteHTTP uses a reverse proxy to route the http request to the scheme, host, and base path provided in target.
+// If the target's path is "/base" and the incoming request was for "/dir",
+// the target request will be for /base/dir.
+func RouteHTTP(ctx context.Context, target string) error {
+	return route(ctx, target, nil)
+}
+
+func route(ctx context.Context, targetHost string, targetPath *string) error {
+	rw := goa.ContextResponse(ctx)
+	if rw == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.New("unable to get response from context"))
+	}
+	req := goa.ContextRequest(ctx)
+	if req == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.New("unable to get request from context"))
+	}
+
+	targetUrl, err := url.Parse(targetHost)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err":         err,
+			"target_host": targetHost,
```

Comment:
add the `target_path` too so that we know which activity was associated with this error.

Reply:
Target path is not very informative here. And it will be nil in most all cases. Added "request_uri": req.RequestURI instead.